Assets/Scripts/Interactables/Harvestable.cs

```diff
@@ -0,0 +1,202 @@
+            { 
+                _gatherTimePassed += Time.deltaTime;
+                if (_gatherTimePassed >= gatherTime)
+                {
+                    AddItems();
+                    if(destroyAfterHarvest) GameObject.Destroy(_parent);
+                    else
+                    {
+                        CoroutineManager.Instance.WaitForSeconds(1.0f/60.0f,() => StartCoroutine(Respawn())); 
+                    }
+                }
+                yield return null;
+            }
+        }
+
+        /// <summary>
+        /// Adds each item with their respective quantity to the player's inventory as long as there is space.
+        /// </summary>
+        private void AddItems()
+        {
+            foreach (KeyValuePair<Item, int> entry in _itemsWithQuantities)
+            {
+                for (int i = 0; i < entry.Value; i++)
+                {
+                    if (!InventoryManager.Instance.AddItem(entry.Key))
+                    {
+                        // Item could not be added to the inventory, drop it to the ground or show an indicator for no space
+                        // For now, just skip to the next item(if there is any) and see if this one can be added
+                        break;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Increases the <see cref="_respawnTimePassed"/> to the limit of <see cref="respawnTime"/>.
+        /// When that limit is reached and the resource is not in the camera of the player,
+        /// it turns its graphics back on.
+        /// </summary>
+        private IEnumerator Respawn()
+        {
+            isRespawning = true;
+            _ownMeshRenderer.enabled = false; 
+            _ownCollider.enabled = false; 
+            _replacementMeshRenderer.enabled = true;
+            while (respawnTimePassed < respawnTime || _ownMeshRenderer.InFrustum(mainCam))
+            {
+                respawnTimePassed += Time.deltaTime;
+                yield return null;
+            }
+            isRespawning = false;
+            respawnTimePassed = 0;
+
+            _ownMeshRenderer.enabled = true;
+            _ownCollider.enabled = true;
+            _replacementMeshRenderer.enabled = false;
+        }
+        
+        // TODO: Code duplication because it's also used in Equipables
+        private List<Equipable.EquipableAbility> GetSelectedElements()
+        {
+            List<Equipable.EquipableAbility> selectedElements = new List<Equipable.EquipableAbility>();
+            for (int i = 0; i < Enum.GetValues(typeof(Equipable.EquipableAbility)).Length; i++)
+            {
+                int layer = 1 << i;
+                if (((int) neededAbility & layer) != 0)
+                {
+                    selectedElements.Add((Equipable.EquipableAbility) i);
+                }
+            }
+
+            return selectedElements;
+        }
```

Comment:
Even better: 
```suggestion
        public static List<Equipable.EquipableAbility> GetSelectedElements(this Equipable.EquipableAbility abilities)
        {
            List<Equipable.EquipableAbility> selectedElements = new List<Equipable.EquipableAbility>();
            for (int i = 0; i < Enum.GetValues(typeof(Equipable.EquipableAbility)).Length; i++)
            {
                int layer = 1 << i;
                if (((int) abilities & layer) != 0)
                {
                    selectedElements.Add((Equipable.EquipableAbility) i);
                }
            }
            
            return selectedElements;
        }
```

Add this to the ClassExtender. Then you should be able to do this:
`_neededAbility.GetSelectedElements()`